app/src/main/java/com/example/habithelper/activities/HabitDetailsActivity.java

```diff
@@ -0,0 +1,147 @@
+
+        tvHabit.setText(habit.getHabitName());
+        if(habit.get("habitImage") != null){
+            ivHabit.setBackground(AppCompatResources.getDrawable(HabitDetailsActivity.this, (Integer) habit.getHabitImageKey()));
+        } else {
+            ivHabit.setBackground(AppCompatResources.getDrawable(HabitDetailsActivity.this, R.drawable.starslarge));
+        }
+
+        btnBack.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent i = new Intent(HabitDetailsActivity.this, MainActivity.class);
+                i.putExtra("tab", "habits");
+                startActivity(i);
+                finish();
+            }
+        });
+    }
+
+    /**
+     * Determines whether the user has been tracking habits long enough to analyze their past data
+     * If so, calls a method to determine and display their most impactful habits
+     * If not, displays a card to tell the user to continue tracking.
+     *
+     */
+    private void getMostImpactfulHabits() {
+        ParseQuery<TrackDay> query = ParseQuery.getQuery(TrackDay.class);
+        query.include(TrackDay.KEY_PARENT_USER);
+        query.include(TrackDay.KEY_DATE_NUMBER);
+        query.include(TrackDay.KEY_TRACK_ARRAY);
+        query.include(TrackDay.KEY_MOOD);
+        query.whereEqualTo(TrackDay.KEY_PARENT_USER, currentUser);
+        query.addDescendingOrder(TrackDay.KEY_DATE_NUMBER);
+        query.findInBackground(new FindCallback<TrackDay>() {
+            @Override
+            public void done(List<TrackDay> daysTracked, ParseException e) {
+                if (e != null) {
+                    return;
+                }
+                numDaysTracked = daysTracked.size();
+
+                // ensures the number of days the user has tracked is greater than the number of habits they are tracking
+                // required for LeastSquaresRegression
+                if (numDaysTracked > numHabits) {
+                    averageMood = getAverageMood(daysTracked);
+                    //TODO: set visibility of percentage calculator!
+                    lrc = new LinearRegressionCalculator();
+                    // calculates the array of correlation coefficients of each habit on the user's mood
+                    lrc.performLeastSquares(daysTracked, numDaysTracked, numHabits);
+                    double[] resultsArray = lrc.leastSquaresResult;
+                    int habitIndex = findIndexOfHabit();
+                    for (int i = 0; i< resultsArray.length; i++){
+                        System.out.println(resultsArray[i]);
+                    }
+                    habitCoefficient = resultsArray[habitIndex];
+                    percentChange = ((int) Math.round(100 * ((double) habitCoefficient / averageMood)));
+                    System.out.println(percentChange);
+                } else {
+                    //TODO: what to do if the user hasn't tracked enough days yet
```

Comment:
Are you tracking this TODO with an issue? If so can you add the issue ID to the comment?

These would help other developers on a bigger project that this TODO is actually tracked and they can check the details or priority o the task by the checking the issue details.

Reply:
I was tracking it - I completed the TODO today but will add issue ID's in the future